src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -270,6 +270,27 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             // Note: persist "m_currentThread" to be compatible with old versions
             info.AddValue(ThreadIDField, 0);
         }
+        
+        [System.Diagnostics.Conditional("DEBUG")]
+        private void DebugDump()
```

Comment:
I would make this return IEnumerable<string>. That way you don't need to limit the number of chunks you return. As you expand in the watch window in the debugger, you see however many you want.

In the body you would use the `yield` keyword

Reply:
As you point out you must then use `#if DEBUG`. Or better still you can conditionally include it in the project using `Condition="'$(Configuration)' == 'Debug'"` .. but nobody else seems to do that so maybe there's a reason not to do that.

Reply:
I would avoid the word Dump in the method name. But this being a debug-only print, maybe it is ok to have it like this?

Reply:
I dont mind what it's called. ShowChunks ?